### PR TITLE
Space Dungeon generation: explicit list of prototypes, name dungeon grids.

### PR DESCRIPTION
**Pull request by whatston3**
## About the PR

Dungeon generation is currently done by enumerating all dungeon configs and removing ones we don't care about.  This turns it into an explicitly whitelisted process, with the AdventureGameRule itself defining the set of space dungeons it generates.

All generated space dungeon grids will be given a name (e.g. "Space Dungeon: VirologyLab").

Haunted and Mineshaft, as they generate using rock walls, were removed from this whitelisted pool - they look odd having rock walls on a giant steel sheet in space.

Note: unsure if the dungeons generate well in Debug configs, but Release seems to work fine.  The space platform itself should probably be removed in favour of having a crappy lattice that the dungeon sits on in space.

## Why / Balance

Missed a dungeon config added upstream containing fultons.

## How to test

1. Start a round, check the admin grid list.
2. Note: the 9 blank entries are still related to this dungeon generation, but I couldn't find where they were created.
3. Warp to one as a ghost, fly outside of it and check the radar, it should not have an IFF name.

## Media

To the right, the admin menu with the space dungeons listed.  To the left, the CaveFactory dungeon.
image

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

No changelog, negligible in-game effect.

### CODE REVIEW

File: Resources/Prototypes/_NF/GameRules/roundstart.yml [hunk 1]
```diff
@@ -4,6 +4,17 @@
...
+    - CaveFactory
+    - MedSci
+    - FactoryDorms
+    - LavaMercenary
+    - VirologyLab
+    - SalvageOutpost
+    # Upstream definitions
+    - Experiment
+    - SnowyLabs
+    - LavaBrig # Skipping Mineshaft, Haunted; they're in caves
```
Comment (dvir001) [inline Resources/Prototypes/_NF/GameRules/roundstart.yml:17]: Can we make the frontier versions just NF ones?

Comment (whatston3) [reply Resources/Prototypes/_NF/GameRules/roundstart.yml:17]: Just redefine it for now?  We're still using these for salvage dungeons as well.

Comment (dvir001) [reply Resources/Prototypes/_NF/GameRules/roundstart.yml:17]: So be it, this is another PR issue.

**Comment by blackknight954** [comment]
So I assume this is the replacement for expeditions?

What is going to end up happening to the expedition shipyard ships

Are the Atmos ships going to just... not be usable?

**Comment by arimah** [comment]
@blackknight954 "Space dungeon" is the internal name for those big dungeon-style wrecks you find in deep space. This PR just fixes them up a bit, and doesn't affect expeditions at all. The future of expeds is still TBD.

**Pull request merged**